### PR TITLE
Update constraints in order to allow some "symfony/*:^5.2" deps

### PR DESCRIPTION
**Pull request by phansys**
## Subject
Update constraints in order to allow some "symfony/*:^5.2" deps.

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Added
- Support for "symfony/asset:^5.2";
- Support for "symfony/config:^5.2";
- Support for "symfony/dependency-injection:^5.2";
- Support for "symfony/http-foundation:^5.2";
- Support for "symfony/templating:^5.2".
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,17 +27,17 @@
...
         "symfony/form": "^4.4",
         "symfony/framework-bundle": "^4.4",
-        "symfony/http-foundation": "^4.4",
+        "symfony/http-foundation": "^4.4 || ^5.2",
         "symfony/http-kernel": "^4.4",
-        "symfony/options-resolver": "^4.4 || ^5.1",
-        "symfony/templating": "^4.4",
+        "symfony/options-resolver": "^4.4 || ^5.2",
+        "symfony/templating": "^4.4 || ^5.2",
         "symfony/twig-bundle": "^4.4",
```
Comment (OskarStark) [inline composer.json:41]: What about this?

Comment (phansys) [reply composer.json:41]: The constraint I left intact are those that belong to dependencies that can't be resolved to version ^5.2.
In this case:
```
composer why-not symfony/twig-bundle:^5.2

sonata-project/admin-bundle  3.91.1     requires          symfony/twig-bundle (^4.4)                      
symfony/twig-bundle          5.2.x-dev  conflicts         symfony/framework-bundle (<5.0)                 
symfony/twig-bundle          5.2.x-dev  conflicts         symfony/translation (<5.0)                      
symfony/twig-bundle          5.2.x-dev  requires          symfony/twig-bridge (^5.0)                      
sonata-project/block-bundle  3.x-dev    does not require  symfony/twig-bridge (but v4.4.20 is installed)  
symfony/twig-bundle          5.2.x-dev  requires          symfony/http-kernel (^5.0)                      
sonata-project/block-bundle  3.x-dev    requires          symfony/http-kernel (^4.4)
```

**Review by VincentLanglet — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thank you @phansys

**Pull request merged**